src/Sylius/Bundle/ApiBundle/EventListener/ChangePaymentMethodEventListener.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\EventListener;
+
+use Sylius\Bundle\ApiBundle\Command\Account\ChangePaymentMethod;
+use Sylius\Bundle\ApiBundle\Command\Checkout\ChoosePaymentMethod;
+use Sylius\Component\Payment\Canceller\PaymentRequestCancellerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+
+final class ChangePaymentMethodEventListener
+{
+    public function __construct(private PaymentRequestCancellerInterface $paymentRequestCanceller)
+    {
+    }
+
+    public function cancelPaymentRequestsWithDifferentPaymentMethod(ControllerArgumentsEvent $event): void
+    {
+        $commands = $event->getArguments();
+        $method = $event->getRequest()->getMethod();
+
+        foreach ($commands as $command) {
+            if ($this->isNotValid($command) || $method !== Request::METHOD_PATCH) {
+                return;
```

Comment:
```suggestion
                continue;
```
Should we check other arguments?